### PR TITLE
feat: Add GraphQL Playground, update PWADevServer

### PR DESCRIPTION
**Pull request by zetlen**
## This PR is a:

[x] New feature
[x] Enhancement/Optimization
[ ] Refactor
[ ] Bugfix
[ ] Test for existing code
[x] Documentation

- Merge DevServerReadyNotifierPlugin into PWADevServer, since they are tightly coupled
- Add GraphQL Playground feature to PWADevServer
- Update PWADevServer docs

### CODE REVIEW

File: packages/pwa-buildpack/docs/PWADevServer.md [hunk 1]
```diff
@@ -77,19 +73,80 @@ configuration.
...
-   backend server, e.g. `'/'`: **Required.** The URL of the backing store.
-- `paths: object`: **Required.** Local absolute paths to project folders.
-  - `output`: Directory for built JavaScript files.
-- `serviceWorkerFileName: string`: **Required.** The name of the ServiceWorker
-   file this project generates, e.g. `'sw.js'`.
-- `changeOrigin: boolean`: ⚠️ **(experimental)** Try to parse any HTML responses
-   from the proxied Magento backend, and replace its domain name with the
-   dev server domain name. Default `false`.
+   backend server, e.g. `'/'`.
+- `provideSecureHost: boolean | SecureHostOptions`: Use a [secure and unique hostname for the dev server.](#securehostoptions)
```
Comment (jcalcaben) [inline packages/pwa-buildpack/docs/PWADevServer.md]: is this option required or optional?

File: packages/pwa-buildpack/docs/PWADevServer.md [hunk 2]
```diff
@@ -77,19 +73,80 @@ configuration.
...
+- `graphqlPlayground: GraphQLPlaygroundOptions`: Add a [pre-populated GraphQL Playground to the dev server.](#graphqlplayground)
```
Comment (jcalcaben) [inline packages/pwa-buildpack/docs/PWADevServer.md]: is this option required or optional?

File: packages/pwa-buildpack/docs/PWADevServer.md [hunk 3]
```diff
@@ -77,19 +73,80 @@ configuration.
...
+  this option to override this behavior and provide the exact domain to use.
+  The custom host will not have a unique hash, but it will have a unique port.
+- `addUniqueHash: boolean`: Use the filesystem path hash to create a short
+  string of URL-safe characters to append to the subdomain. Ensures total
+  uniqueness of domain. Default `true`. If `exactDomain` is specified, this
+  option has no effect.
+
+:information_source: Default behavior is to use the `package.json` plus hash
+as a subdomain of `local.pwadev`. To use this default behavior, simply set
+`provideUniqueHost: true`.
```
Comment (jcalcaben) [inline packages/pwa-buildpack/docs/PWADevServer.md]: Is this supposed to be `provideSecureHost` instead of `provideUniqueHost`?

Comment (zetlen) [reply packages/pwa-buildpack/docs/PWADevServer.md]: Yes, whoops!

File: packages/pwa-buildpack/docs/PWADevServer.md (hunk 2 above)
Comment (zetlen) [reply packages/pwa-buildpack/docs/PWADevServer.md]: Both are optional.

File: packages/pwa-buildpack/docs/PWADevServer.md (hunk 3 above)
Comment (zetlen) [reply packages/pwa-buildpack/docs/PWADevServer.md]:
```suggestion
`provideSecureHost: true`.
```

Trying the new suggestions feature...

**Pull request merged**